### ISSUE TITLE
2 bugs that crash the game

### ISSUE THREAD

**Issue by numberZero** [sapier/animals_modpack#109]
2 bugs that cause crash:
1. in mobf_settings/fstk/ui_mod.lua function minetest.register_on_leaveplayer is called with undefined global variable "player" as parameter. Perhaps it should be function "player_leave"?
2. function fighting.update_healdb from mobf/fighting.lua sometimes gets called with item name passed as replace_with_item, and sometimes with item definition.

Quick workaround (git format-patch):
```
From 9a5daa72c144b6f2aa3853b7585d0f0cd0828125 Mon Sep 17 00:00:00 2001
From: number Zero <silverunicorn2011@yandex.ru>
Date: Sun, 27 Sep 2015 23:51:17 +0300
Subject: [PATCH] Bugfix (2 crash reasons)

---
 mobf/fighting.lua             | 4 +++-
 mobf_settings/fstk/ui_mod.lua | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/mobf/fighting.lua b/mobf/fighting.lua
index cdca9a0..bd6921d 100644
--- a/mobf/fighting.lua
+++ b/mobf/fighting.lua
@@ -1380,7 +1380,9 @@ function fighting.update_healdb(hp_change, replace_with_item, itemstack, player,
 	
 	local replacement = nil
 	if replace_with_item ~= nil then
-		replacement = replace_with_item:get_name()
+		if type(replace_with_item) ~= "string" then
+			replacement = replace_with_item:get_name()
+		end
 	end
 	
 	if fighting.healdb == nil then
diff --git a/mobf_settings/fstk/ui_mod.lua b/mobf_settings/fstk/ui_mod.lua
index 5e85e78..2242346 100644
--- a/mobf_settings/fstk/ui_mod.lua
+++ b/mobf_settings/fstk/ui_mod.lua
@@ -200,4 +200,4 @@ end
 --------------------------------------------------------------------------------
 
 minetest.register_on_player_receive_fields(handle_buttons)
-minetest.register_on_leaveplayer(player)
+minetest.register_on_leaveplayer(player_leave)
-- 
2.5.3
```

**Comment by svenXY** [comment]
I can confirm that it fixes the bug. Thanks

**Comment by sapier** [comment]
fixed in 0.4.93

**Comment by asziranyi** [comment]
Cool, thanks @sapier!
